bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/api/CatalogQuery.scala

```diff
@@ -1,31 +1,31 @@
 package edu.gemini.catalog.api
 
-import edu.gemini.spModel.core.{Magnitude, Coordinates}
+import edu.gemini.spModel.core.Coordinates
 import edu.gemini.spModel.core.Target.SiderealTarget
 
 import scalaz._
 import Scalaz._
 
+trait QueryResultsFilter {
+  def filter(t: SiderealTarget): Boolean
+}
+
+case class RadiusFilter(base: Coordinates, rc: RadiusConstraint) extends QueryResultsFilter {
+  def filter(t: SiderealTarget): Boolean = rc.targetsFilter(base)(t)
+}
+
+case class MagnitudeQueryFilter(filters: NonEmptyList[MagnitudeConstraints]) extends QueryResultsFilter {
+  def filter(t: SiderealTarget): Boolean = filters.list.forall(_.filter(t))
+}
+
```

Comment:
I think I might have made a single `CatalogQuery` case class with base position, radius constraint, and magnitude constraints.  You can do the filtering in your `filter` method based on those.  I'm not sure I follow what is going on with the separate `QueryResultsFilter`, `RadiusFilter`, `MagnitudeQueryFilter`, `BandConstrainedCatalogQuery`, `RangeConstrainedCatalogQuery` and the various constructors.  What do they buy us?  `CatalogQuery` is `sealed` so as a user of this API there's no way to define and apply your own filters anyway?